Do-It/Controller/DueDatePickerTableViewController.swift

```diff
@@ -0,0 +1,155 @@
+    struct Option {
+
+        let index: Int
+        let name: String
+        let dateRange: DateInterval
+
+        init(row: Int, description: String, range: DateInterval) {
+            index = row
+            name = description
+            dateRange = range
+        }
+    }
+
+    var options: [Option] = []
+
+    weak var delegate: DueDatePickerTableViewControllerDelegate?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        makeTimeOptions()
+    }
+
+    func makeTimeOptions() {
+
+        var components = DateComponents()
+        components.day = 1
+        components.second = -1
+
+        let startOfDay = Calendar.current.startOfDay(for: Date())
+        let endOfDay = Calendar.current.date(byAdding: components, to: startOfDay)!
+        let endOfTomorrow = Calendar.current.date(byAdding: .day, value: 1, to: endOfDay)!
+
+        options.append(Option(row: 0,
+                              description: "Today",
+                              range: DateInterval(start: Date(), end: endOfDay)))
+        options.append(Option(row: 1,
+                              description: "Tomorrow",
+                              range: DateInterval(start: endOfDay, end: endOfTomorrow)))
+        options.append(Option(row: 2,
+                              description: "Within the next Three Days",
+                              range: DateInterval(start: Date(),
+                                                  end: Calendar.current.date(byAdding: .day, value: 3, to: Date())!)))
+        options.append(Option(row: 3,
+                              description: "Within the next Week",
+                              range: DateInterval(start: Date(),
+                                                  end: Calendar.current.date(byAdding: .day, value: 7, to: Date())!)))
+
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return options.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DueDateCell", for: indexPath)
+        cell.textLabel?.text = options[indexPath.item].name
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
+        if let cell = tableView.cellForRow(at: indexPath) {
+            cell.accessoryType = .none
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        timeRangeSelected = options[indexPath.row].dateRange
+        if let cell = tableView.cellForRow(at: indexPath) {
```

Comment:
As a side-effect of assigning the checkbox in `tableView(_:cellForRowAt)`, we don't have to do this explicitly here. You can remove `tableView(_:didDeselectRowAt:)` entirely, and in this function after changing `timeRangeSelected`, just call `tableView.reloadRows(at:)` and pass the index path that was selected.